### PR TITLE
Feature/creating heading field for components

### PR DESCRIPTION
**Pull request by katlazarova**

### CODE REVIEW

File: web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    not paragraph.isPublished() ? 'paragraph--unpublished'
+  ]
+%}
+{% block paragraph %}
+  <div{{ attributes.addClass(classes) }}>
+    {% include "@components/_heading.twig" with {
+      heading_level: 2,
+      heading_text: content.field_p_c_heading|field_value|first['#context'].value,
+      heading_attributes: create_attribute().addClass(content.field_p_c_visually_hide_heading|field_value|first['#markup'] == 'On' ? 'visually-hidden')
+    } %}
```
Comment (ElXyfer) [inline web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig:55]: When doing comparative checks like this it's better to do ```[entity_name].field_name.value``` where entity name is paragraph or node for example

**Comment by ElXyfer** [comment]
Components paragraph needs to be renamed to be Component.

**Comment by katlazarova** [comment]
> Components paragraph needs to be renamed to be Component.

it is

File: web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig [hunk 2]
```diff
@@ -38,5 +38,5 @@
  */
 #}
 {% include "@components/_field.twig" with {
-  attributes: create_attribute().addClass('u-o-3')
```
Comment (ElXyfer) [inline web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig:41]: This change now means there's a spacing inconsistency between the basic pages and the recipes/articles. Is this deliberate? An argument can be made for either case

File: web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    view_mode ? 'paragraph--view-mode--' ~ view_mode|clean_class,
+    not paragraph.isPublished() ? 'paragraph--unpublished'
+  ]
+%}
+{% block paragraph %}
+  <div{{ attributes.addClass(classes) }}>
+    {% include "@components/_heading.twig" with {
+      heading_level: 2,
+      heading_text: content.field_p_c_heading|field_value|first['#context'].value,
+      heading_attributes: create_attribute().addClass(content.field_p_c_visually_hide_heading|field_value|first['#markup'] == 'On' ? 'visually-hidden')
```
Comment (ElXyfer) [inline web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig:54]: This should not be rendered if the heading is empty. Right now if no heading is added an empty h2 is rendered.

File: web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig (hunk 1 above)
Comment (katlazarova) [reply web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig:55]: Nvm, thank you!

File: web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig (hunk 2 above)
Comment (katlazarova) [reply web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig:41]: It is deliberate actually. The spacing increase is not necessary on the article and recipe content types which tend to have a different structure and are heavily text based with any additional components just adding to the main content of the page. Basic pages are more varied and the components are more likely to be unrelated and require more spacing between them

File: web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig (hunk 3 above)
Comment (katlazarova) [reply web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig]: fixed.

File: web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig (hunk 2 above)
Comment (ElXyfer) [reply web/themes/custom/pineapple/templates/field/field--field-bp-body.html.twig:41]: Okay No problem.

File: web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig [hunk 4]
```diff
@@ -48,11 +48,13 @@
 %}
 {% block paragraph %}
   <div{{ attributes.addClass(classes) }}>
-    {% include "@components/_heading.twig" with {
-      heading_level: 2,
-      heading_text: content.field_p_c_heading|field_value|first['#context'].value,
-      heading_attributes: create_attribute().addClass(paragraph.field_p_c_visually_hide_heading.value == 1 ? 'visually-hidden')
-    } %}
+    {% if content.field_p_c_heading|field_value %}
```
Comment (ElXyfer) [inline web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig:51]: Checks like this should be done like "not entity.field_name.isEmpty()"

Comment (katlazarova) [reply web/themes/custom/pineapple/templates/paragraphs/paragraph--components.html.twig]: Done.

**Pull request merged**